sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveUDFs.scala

```diff
@@ -293,69 +307,57 @@ private[hive] case class HiveUDAFFunction(
   private lazy val inspectors = children.map(toInspector).toArray
 
   @transient
-  private lazy val functionAndInspector = {
+  private lazy val function = {
     val parameterInfo = new SimpleGenericUDAFParameterInfo(inspectors, false, false)
-    val f = resolver.getEvaluator(parameterInfo)
-    f -> f.init(GenericUDAFEvaluator.Mode.COMPLETE, inspectors)
+    resolver.getEvaluator(parameterInfo)
   }
 
   @transient
-  private lazy val function = functionAndInspector._1
+  private lazy val wrappers = children.map(x => wrapperFor(toInspector(x), x.dataType)).toArray
 
   @transient
-  private lazy val wrappers = children.map(x => wrapperFor(toInspector(x), x.dataType)).toArray
+  private lazy val returnInspector =
+    function.init(GenericUDAFEvaluator.Mode.COMPLETE, inspectors)
 
   @transient
-  private lazy val returnInspector = functionAndInspector._2
+  private lazy val partialResultInspector =
+    function.init(GenericUDAFEvaluator.Mode.PARTIAL1, inspectors)
+
+  // The following two lines initializes `function: GenericUDAFEvaluator` eagerly. These two fields
```

Comment:
This feels little weird. If you want eager init and not serialize, why not remove `lazy` and just keep it `@transient` ?